agent/helpers.go

```diff
@@ -98,11 +97,12 @@ func (h Helper) isRouteExist(ip net.IP, netmask string) error {
 }
 
 // createRoute creates IP route, returns nil if success and error otherwise.
-func (h Helper) createRoute(ip net.IP, netmask string, via string, dest string) error {
+func (h Helper) createRoute(ip net.IP, netmask string, via string, dest string, extraArgs ...string) error {
```

Comment:
I would be happier with a "source string" parameter and if left empty then no source is specified. The way it is right now, knowledge about the exact syntax of the "ip" command has to leak out into "ensureRouteToEndpoint", when the whole idea of "createRoute" was that it would take care of those details and hide "ip" related route-creation syntax.